### DISCUSSION TITLE
设置outbounds后，访问相关网站没有效果

### DISCUSSION THREAD

**Discussion by hellotern** [XTLS/Xray-core#2022] [General]
背景：由于我在AWS上购买了一个虚拟机，但无法访问chat.openai.com网站，所有就用cloudflare warp包装一下，同时安装了xray。

配置文件config.json：
{
"outbounds": [
    {
        "tag": "chatgpt_proxy",
        "protocol": "socks",
        "settings": {
            "servers": [
                {
                    "address": "127.0.0.1",
                    "port": 40000
                }
            ]
        }
    }
],
"routing": {
    "rules": [
        {
            "type": "field",
            "outboundTag": "chatgpt_proxy",
            "domain": [
                "openai.com",
		"ifconfig.me"
            ],
            "enabled": true
        }
    ]
  }
}

xray服务启动都是正常的

当我在终端输入：curl ifconfig.me，返回的ip地址是虚拟机的公网IP地址，但当我在终端输入：curl ifconfig.me --proxy socks5://127.0.0.1:40000，返回的是warp之后的ip地址。我现在想要达到的目的是输入：curl ifconfig.me，也应该是返回warp之后的ip地址，但没有看到效果，感觉应该是配置文件哪个地方配置的不对，不知哪位是否碰到过类似的问题，谢谢了

**Comment by avenger2005** [comment]
你应该是设置了warp的sock5代理模式，监听40000端口，而非全局出站。
所以在vps内部curl 的话，不指定代理，会直接出站，返回是vps的ip。
如果在浏览器通过vps访问，xray会接管再转warp，这个时候会返回warp的ip。
